agent/structs/connect_ca.go

```diff
@@ -460,6 +461,13 @@ type VaultCAProviderConfig struct {
 	TLSSkipVerify bool
 }
 
+type AWSCAProviderConfig struct {
```

Comment:
I think you are missing the mirror change in `api`.

Reply:
Right now we don't have structs in `api` for provider-specific configs since they are just a `map[string]interface{}` There is one for the built in provider but not sure why honestly.

The one thing I did miss there though in my last PR was adding the State field which is now in the response although can't be set in a request. I should also validate that the user isn't writing it directly in Config PUT too I suppose.

Reply:
^ PUT validation for no state modification done locally.